@Twixes, I tried to run the command `bin/e2e-test-runner`, but I got the error below. The same script works fine on master. Did I do something wrong?

```
2024-07-30 11:24:43 [warning  ] ️WARNING! Environment variable E2E_TESTING is enabled. This is a security vulnerability unless you are running tests.
{'event': ['Skipping service version requirements. This is dangerous and PostHog might not work as expected!'], 'timestamp': '2024-07-30T09:24:43.758175Z', 'logger': 'posthog.settings.service_requirements', 'level': 'warning', 'pid': 22167, 'tid': 8423623680}
2024-07-30T09:24:46.639066Z [warning  ] Skipping async migrations setup. This is unsafe in production! [posthog.apps] pid=22167 tid=8423623680
2024-07-30T09:24:46.640829Z [info     ] AXES: BEGIN LOG                [axes.apps] pid=22167 tid=8423623680
2024-07-30T09:24:46.641167Z [info     ] AXES: Using django-axes version 5.9.0 [axes.apps] pid=22167 tid=8423623680
2024-07-30T09:24:46.641210Z [info     ] AXES: blocking by IP only.     [axes.apps] pid=22167 tid=8423623680
Traceback (most recent call last):
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 366, in _get_applied_migrations_and_create_tables
    return self._get_applied_migrations(migrations_package_name, replicated)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 383, in _get_applied_migrations
    return set(obj.module_name for obj in self.select(query))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 383, in <genexpr>
    return set(obj.module_name for obj in self.select(query))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 280, in select
    r = self._send(query, settings, True)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 393, in _send
    raise ServerError(r.text)
infi.clickhouse_orm.database.ServerError: Code: 60. DB::Exception: Table posthog_test.infi_clickhouse_orm_migrations_distributed does not exist. Maybe you meant default.infi_clickhouse_orm_migrations_distributed?. (UNKNOWN_TABLE) (version 23.12.5.81 (official build))
 (0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ggg/Projects/posthog/posthog/manage.py", line 22, in <module>
    main()
  File "/Users/ggg/Projects/posthog/posthog/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/posthog/management/commands/migrate_clickhouse.py", line 53, in handle
    self.migrate(CLICKHOUSE_HTTP_URL, options)
  File "/Users/ggg/Projects/posthog/posthog/posthog/management/commands/migrate_clickhouse.py", line 92, in migrate
    database.migrate(MIGRATIONS_PACKAGE_NAME, options["upto"], replicated=True)
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 350, in migrate
    applied_migrations = self._get_applied_migrations_and_create_tables(migrations_package_name, replicated=replicated)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 370, in _get_applied_migrations_and_create_tables
    self.create_table(MigrationHistoryReplicated)
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 156, in create_table
    self._send(model_class.create_table_sql(self))
  File "/Users/ggg/Projects/posthog/posthog/env/lib/python3.11/site-packages/infi/clickhouse_orm/database.py", line 393, in _send
    raise ServerError(r.text)
infi.clickhouse_orm.database.ServerError: Code: 253. DB::Exception: There was an error on [localhost:9000]: Code: 253. DB::Exception: Cannot create table, because it is created concurrently every time or because of wrong zookeeper_path or because of logical error. (REPLICA_ALREADY_EXISTS) (version 23.12.5.81 (official build)). (REPLICA_ALREADY_EXISTS) (version 23.12.5.81 (official build))
 (0)
[1]    22141 terminated  bin/e2e-test-runner
```

--- a/.run/Celery Threads.run.xml
+++ b/.run/Celery Threads.run.xml
@@ -21,7 +21,7 @@
     <option name="ADD_SOURCE_ROOTS" value="true" />
     <EXTENSION ID="PythonCoverageRunConfigurationExtension" runner="coverage.py" />
     <option name="SCRIPT_NAME" value="$PROJECT_DIR$/manage.py" />
-    <option name="PARAMETERS" value="run_autoreload_celery" />
+    <option name="PARAMETERS" value="run_autoreload_celery --type=worker" />
     <option name="SHOW_COMMAND_LINE" value="false" />
     <option name="EMULATE_TERMINAL" value="false" />
     <option name="MODULE_MODE" value="false" />

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -94,7 +94,7 @@
             },
             "request": "launch",
             "program": "${workspaceFolder}/manage.py",
-            "args": ["run_autoreload_celery"],
+            "args": ["run_autoreload_celery", "--type=worker"],
             "console": "integratedTerminal",
             "python": "${workspaceFolder}/env/bin/python",
             "cwd": "${workspaceFolder}",

--- a/bin/e2e-test-runner
+++ b/bin/e2e-test-runner
@@ -46,15 +46,14 @@ do
     fi
 done
 
-export DEBUG=1
 export NO_RESTART_LOOP=1
 export CYPRESS_BASE_URL=http://localhost:8080
 export SECURE_COOKIES=0
 export SKIP_SERVICE_VERSION_REQUIREMENTS=1
 export KAFKA_HOSTS=kafka:9092
 export CLICKHOUSE_DATABASE=posthog_test
-export TEST=1 # Plugin server and kafka revert to 'default' Clickhouse database if TEST is not set
 export CLICKHOUSE_SECURE=0
+export JS_URL=http://localhost:8234
 export E2E_TESTING=1
 export SECRET_KEY=e2e_test
 export EMAIL_HOST=email.test.posthog.net

--- a/bin/start-worker
+++ b/bin/start-worker
@@ -7,8 +7,8 @@ trap 'kill $(jobs -p)' EXIT
 source ./bin/celery-queues.env
 
 # start celery worker with heartbeat (-B)
-SKIP_ASYNC_MIGRATIONS_SETUP=0 CELERY_WORKER_QUEUES=$CELERY_WORKER_QUEUES celery -A posthog worker --without-heartbeat --without-mingle --pool=threads -Ofair -n node@%h &
-celery -A posthog beat -S redbeat.RedBeatScheduler &
+python manage.py run_autoreload_celery --type=worker &
+python manage.py run_autoreload_celery --type=beat &
 
 if [[ "$PLUGIN_SERVER_IDLE" != "1" && "$PLUGIN_SERVER_IDLE" != "true" ]]; then
   ./bin/plugin-server

--- a/cypress.e2e.config.ts
+++ b/cypress.e2e.config.ts
@@ -99,15 +99,16 @@ export default defineConfig({
                     const redisClient = await createClient()
                         .on('error', (err) => console.log('Redis client error', err))
                         .connect()
-                    for await (const key of redisClient.scanIterator({
-                        TYPE: 'string',
-                        MATCH: '*cache*',
-                        COUNT: 100,
-                    })) {
+                    // Clear cache
+                    for await (const key of redisClient.scanIterator({ TYPE: 'string', MATCH: '*cache*', COUNT: 500 })) {
+                        await redisClient.del(key)
+                    }
+                    // Also clear the more ephemeral async query statuses
+                    for await (const key of redisClient.scanIterator({ TYPE: 'string', MATCH: 'query_async*', COUNT: 500 })) {
                         await redisClient.del(key)
                     }
                     await redisClient.quit()
-                    return null
+                    return null // Cypress requires _some_ return value
                 },
             })
 

--- a/posthog/clickhouse/table_engines.py
+++ b/posthog/clickhouse/table_engines.py
@@ -49,7 +49,7 @@ class MergeTreeEngine:
             shard_key, replica_key = "noshard", "{replica}-{shard}"
 
         # ZK is not automatically cleaned up after DROP TABLE. Avoid zk path conflicts in tests by generating unique paths.
-        if settings.TEST and self.zookeeper_path_key is None or self.force_unique_zk_path:
+        if (settings.TEST or settings.E2E_TESTING) and self.zookeeper_path_key is None or self.force_unique_zk_path:
             self.set_zookeeper_path_key(str(uuid.uuid4()))
 
         if self.zookeeper_path_key is not None:

--- a/posthog/management/commands/run_autoreload_celery.py
+++ b/posthog/management/commands/run_autoreload_celery.py
@@ -1,3 +1,4 @@
+from typing import Literal
 import django
 from django.core.management.base import BaseCommand
 
@@ -10,26 +11,34 @@ from django.utils import autoreload  # noqa: E402
 
 
 class Command(BaseCommand):
-    help = "Wrap celery in djangos auto reload functionality"
+    help = "Run Celery wrapped in Django's auto-reload feature"
+
+    def add_arguments(self, parser):
+        parser.add_argument("--type", type=str, choices=("worker", "beat"), help="Process type")
 
     def handle(self, *args, **options):
-        autoreload.run_with_reloader(self.run_celery_worker)
+        autoreload.run_with_reloader(lambda: self.run_celery_worker(options["type"]))
 
-    def run_celery_worker(self):
+    @staticmethod
+    def run_celery_worker(type: Literal["worker", "beat"]):
         from posthog.celery import app as celery_app
 
-        queues = [q.value for q in CeleryQueue]
-
-        args = [
-            "-A",
-            "posthog",
-            "worker",
-            "-B",
-            "--pool=threads",
-            f"--queues={','.join(queues)}",
-            "-Ofair",
-            "-n",
-            "node@%h",
-        ]
+        args = (
+            [
+                "-A",
+                "posthog",
+                "worker",
+                "--pool=threads",
+                f"--queues={','.join(q.value for q in CeleryQueue)}",
+            ]
+            if type == "worker"
+            else [
+                "-A",
+                "posthog",
+                "beat",
+                "--scheduler",
+                "redbeat.RedBeatScheduler",
+            ]
+        )
 
         celery_app.worker_main(args)

